src/main/kotlin/view/InputView.kt

```diff
@@ -0,0 +1,104 @@
+package view
+
+import model.Lotto
+
+class InputView {
+    private fun readLineOrRetry(prompt: String): String {
+        while (true) {
+            print(prompt)
+            val input = readLine()
+            if (input != null) {
+                return input
+            } else {
+                println(RETRY_INPUT_MESSAGE)
+            }
+        }
+    }
+
+    private fun parsePurchaseAmount(input: String): Int? {
+        return input.toIntOrNull()
+    }
+
+    fun getPurchaseAmount(): Int {
+        while (true) {
+            val input = readLineOrRetry(GET_PURCHASE_AMOUNT)
+            val amount = parsePurchaseAmount(input)
+            if (amount != null) {
+                return amount
+            } else {
+                println(INVALID_INPUT_MESSAGE)
+            }
+        }
+    }
+
+    private fun convertWinningNumbers(input: String): List<Int> {
+        return try {
+            input.split(",")
+                .map { it.trim().toInt() }
```

Comment:
```suggestion
            input.split(",")
                .map(String::trim)
                .map(String::toInt)
```

Kotlin offers A LOT of ways for us to do things, and readability doesn't always have one correct answer.
So this review is just a suggestion. Which one looks better?

Reply:
I thought about my version and suggested version. Since I'm used to C programming, I thought my version has some pros using slightly less memory.

But I ended up thinking that pros of suggested version is more important, since...
1. it has clear separation of tasks
2. it improved readability
3. it's easier to debug

So I would use your version in Kotlin project. Thanks for suggesting me the new version!